airflow-core/src/airflow/notifications/basenotifier.py

```diff
@@ -0,0 +1,22 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+"""Re exporting the new notifier module from Task SDK for backward compatibility."""
+
+from __future__ import annotations
+
+from airflow.sdk.bases.notifier import BaseNotifier as BaseNotifier
```

Comment:
I think importing this module should trigger a compatibility warning, Saying something like "will be removed in Airflow 3.1")

Reply:
Yes we should use the dynamic deprecation code like in other places (utils/deprecations? )

Reply:
Fair ask, let me try that